### PR TITLE
[thing] Dynamic state/command provider should not return original description 

### PR DESCRIPTION
**Pull request by cweitkamp**
- Dynamic state/command provider should not return original description
- Added integration test for `CommandDescriptionProvider`

See https://community.openhab.org/t/dynamic-state-options-not-showing-up/95625

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ public Integer getRank() {
             @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
         for (DynamicStateDescriptionProvider provider : dynamicStateDescriptionProviders) {
             StateDescription stateDescription = provider.getStateDescription(channel, originalStateDescription, locale);
-            if (stateDescription != null) {
+            if (stateDescription != null && !stateDescription.equals(originalStateDescription)) {
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: Question is if we should just ignore these wrong implementations in bindings or if we should throw an exception here and force the developer to fix it.

Comment (mhilbush) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: Being completely silent about it doesn't feel right. Would the exception be thrown every time the state options are accessed? If so, that could result in a lot of exceptions. Maybe just log it instead?

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: I added a ERROR log message.

Comment (wborn) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: Can you also update the JavaDocs of these providers for the fact that if the result is not null it should not be the original object?

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: I already did (see e.g. [here](https://github.com/openhab/openhab-core/pull/1399/files#diff-af4592feefd1fdeb2cb4cfcc431d75c0R38)).

Comment (wborn) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ChannelStateDescriptionProvider.java]: Great! Also nice to see some unit tests for this. :+1:

**Pull request merged**